config/initializers/zhe_badges.rb

```diff
@@ -0,0 +1,3 @@
+Dir[Rails.root + './app/badges/*.rb'].each do |file|
+  require file
+end
```

Comment:
Exactly, this file must be loaded after devise initializer, this is important for `on` method casting on `User` model. In the future, there can be problems with other initializers.